### PR TITLE
Migrate QRegExp to QRegularExpression

### PR DESCRIPTION
**Pull request by har13205**
# Description

PR to migrate `QRegExp` to `QRegularExpression`

## Type of change

- [ ] Bug fix
- [ ] New sample implementation
- [x] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [ ] Windows
- [ ] Android
- [ ] Linux
- [x] macOS
- [ ] iOS

## Checklist

- [ ] Runs and compiles on all active platforms as a standalone sample
- [ ] Runs and compiles in the sample viewer(s)
- [ ] Branch is up to date with the latest main/v.next
- [ ] All merge conflicts have been resolved
- [ ] Self-review of changes
- [ ] There are no warnings related to changes
- [ ] No unrelated changes have been made to any other code or project files
- [ ] Code is commented with correct formatting (CTRL+i)
- [ ] All variable and method names are camel case
- [ ] There is no leftover commented code
- [ ] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [ ] If adding a new sample, it is added to the sample viewer
- [ ] Cherry-picked to Main branch (if applicable)

### CODE REVIEW

File: SampleViewer/SyntaxHighlighter/QMLHighlighter.cpp [hunk 1]
```diff
@@ -91,49 +91,52 @@ QMLHighlighter::QMLHighlighter(QTextDocument* parent) :
...
     if (previousBlockState() != 1)
-        startIndex = m_commentStartExpression.indexIn(text);
+        startIndex = m_commentStartExpression.match(text).capturedStart();
 
     while (startIndex >= 0)
     {
-        int endIndex = m_commentEndExpression.indexIn(text, startIndex);
+        QRegularExpressionMatch match = m_commentEndExpression.match(text, startIndex);
+
+        int endIndex = match.capturedStart();
```
Comment (jared-2016) [inline SampleViewer/SyntaxHighlighter/QMLHighlighter.cpp:139]: Just one comment.(Optional) This is the only instance where we break it up into a `QRegularExpressionMatch` then get the `index` afterwards. It doesn't look like we reuse `match` afterwards besides getting the index, so maybe just simplify it to one line like the rest?

File: SampleViewer/SyntaxHighlighter/QMLHighlighter.cpp
Comment (JamesMBallard) [inline SampleViewer/SyntaxHighlighter/QMLHighlighter.cpp:1]: The file should be renamed to QmlHighlighter.cpp .

File: SampleViewer/SyntaxHighlighter/QMLHighlighter.h
Comment (JamesMBallard) [inline SampleViewer/SyntaxHighlighter/QMLHighlighter.h:1]: file should be renamed to match new references `QmlHighlighter.h`

**Review by JamesMBallard — COMMENTED** [review]
You need to consolidate the

**Review by JamesMBallard — APPROVED** [review]

**Review by GuillaumeBelz — APPROVED** [review]

**Pull request merged**